### PR TITLE
Log stacktraces in SystemOutLogger

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description
Log stacktraces in SystemOutLogger when exception occurs.

## :bulb: Motivation and Context

With `debug` mode on, when exception happens it's currently difficult to find out why and where did it happen. With stacktraces printed to the console we can easily find the bug.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: sentry-core/src/main/java/io/sentry/core/SystemOutLogger.java [hunk 1]
```diff
@@ -42,6 +48,18 @@ public void log(SentryLevel level, String message, Throwable throwable) {
   @Override
   public void log(SentryLevel level, Throwable throwable, String message, Object... args) {
     System.out.println(
-        String.format("%s: %s \n %s", level, String.format(message, args), throwable.toString()));
+        String.format(
```
Comment (bruno-garcia) [inline sentry-core/src/main/java/io/sentry/core/SystemOutLogger.java:51]: Makes me realize the previous code was a bit naive but don't we need to check if we have an instance in the first place?
I'm afraid we might throw if any arg is null here

Comment (maciejwalkowiak) [reply sentry-core/src/main/java/io/sentry/core/SystemOutLogger.java:51]: True! Good catch. Now I see that `throwable` is marked as `@Nullable` in `DiagnosticLogger#log`. I'll take care of it.

File: sentry-core/src/main/java/io/sentry/core/SystemOutLogger.java [hunk 2]
```diff
@@ -41,7 +51,23 @@ public void log(SentryLevel level, String message, Throwable throwable) {
...
+          String.format(
+              "%s: %s \n %s\n%s",
+              level,
+              String.format(message, args),
+              throwable.toString(),
+              captureStackTrace(throwable)));
+    }
+  }
+
+  private String captureStackTrace(@NotNull Throwable throwable) {
```
Comment (marandaneto) [inline sentry-core/src/main/java/io/sentry/core/SystemOutLogger.java:67]:
```suggestion
  private @NotNull String captureStackTrace(final @NotNull Throwable throwable) {
```

File: sentry-core/src/main/java/io/sentry/core/SystemOutLogger.java (hunk 1 above)
Comment (marandaneto) [reply sentry-core/src/main/java/io/sentry/core/SystemOutLogger.java]: good catch, we've never used this class, that's why not so feature-complete (missing stack traces, etc...), we're always in favor of AndroidLogger for Android.

**Pull request merged**